sdk/rust/oak/tests/handle_extract_inject.rs

```diff
@@ -0,0 +1,267 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+include!(concat!(env!("OUT_DIR"), "/handle_tests/tests.rs"));
+
+use oak::{
+    handle::{extract_handles, inject_handles},
+    io::{Receiver, Sender},
+};
+
+#[test]
+fn extract_nothing() {
+    let mut message = LookMaNoHandles {
+        a: "Hello, world!".to_string(),
+        b: 42,
+    };
+
+    let handles = extract_handles(&mut message);
+
+    assert_eq!(handles, vec![]);
+}
+
+#[test]
+fn extract_struct() {
+    let mut message = TestMessage {
+        other_arbitrary_field: "Test".to_string(),
+        test_sender: Some(sender(42)),
+        test_receiver: Some(receiver(1337)),
+    };
+
+    let handles = extract_handles(&mut message);
+
+    assert_eq!(handles, vec![42, 1337]);
+    assert_eq!(
+        message,
+        TestMessage {
+            other_arbitrary_field: "Test".to_string(),
+            test_sender: Some(sender(0)),
+            test_receiver: Some(receiver(0)),
+        }
+    );
```

Comment:
Would round-trip tests (message->extract->inject->same message) be worthwhile?

Reply:
I suppose it wouldn't hurt to add a few spot checks doing this 👍

Reply:
I think having a table-driven test (as opposed to separate test cases) for this would be great, so we can easily add examples that we expect to round-trip correctly.

Reply:
So table-driven is slightly complicated because I can't make a `Vec<HandleVisit>` due to the `visit` function being generic 😅. 
I've now opted to just have one container proto that contains all other test messages, and do the roundtrip on that (will push those changes once everything builds).